chain/network-primitives/src/config_json.rs

```diff
@@ -125,8 +131,7 @@ pub struct Config {
     /// If the list is empty and STUN servers' response is ambiguous, the peers which connect to
     /// this validator node will natually observe the address of the validator and broadcast it.
     /// This setup is not reliable in presence of byzantine peers.
-    /// TODO: unskip, once the functionality is implemented.
-    #[serde(skip)]
+    #[serde(default)]
     pub public_addrs: Vec<String>,
```

Comment:
Can use `SocketAddr` here instead of String? 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5169a158bb24513f6b32ef6ddf6f683a

Reply:
yes and no: eventually I want to support <peer>@<IP>:<port> (see TODO above) so then we will be back to Vec<String> anyway.

Reply:
For that todo, I'd imagine we do `Vec<SocketAddrWithKey>`? Ie, introduce a strongly-typed struct here anyway, which would properly parse the format during deserialization

Reply:
We have a separate of abstraction layer for strongly typed configs, but sure, we can have this here as well.

Reply:
I think `NetworkConfig` and such are not as much for adding strong types, as they are for cutting dependencies between components. In particular, `NetworkConfig::new` is infailable, so it can't report new errors.

My general rule-of-thumb is to push as much validation as possible to the serialization layer, but not more. That is:

* strong preference to validating syntactic properties in Serialize impl
* moderate preference for restricting all properties to be syntactic 
* strong preference for keeping semantic validation out of the config.

Reply:
done

Reply:
This is a bit philosophical, let's chat about this!